src/main.cpp

```diff
@@ -588,6 +586,15 @@ int main(int argc, char **argv)
 		cerr << "--hsm-module missing but --hsm-so-pin and/or --hsm-pin given" << endl;
 		return EXIT_FAILURE;
 	}
+#ifdef AKLITE_TAGS
+	if (options.pacman_tags.empty()) {
+		if (!boost::filesystem::exists(os_release_filename) &&
+			!boost::filesystem::exists(installed_versions_filename)) {
+			cerr << "Default tag missing. Please use --tags to register the device." << endl;
+			return EXIT_FAILURE;
```

Comment:
maybe is better to move this down to take into account the parsing result.

Reply:
Or better to move the parsing up. If the tags are not set, there is no point making a http request to the DG.

Reply:
looks goods as well

Reply:
>if the tags are not set, there is no point making a http request to the DG.
Agree. We should parse the files before any requests, and if no success in the tag parsing then exit with error.

Maybe it makes sense to encapsulate the new functionality in a function.

Reply:
And then do something like
```
("tags,t", po::value<string>(&options.pacman_tags)->default_value(get_default_tag()),
```
then below do `_assert_tags(options.tags)`